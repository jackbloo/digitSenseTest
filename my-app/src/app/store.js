import { configureStore } from '@reduxjs/toolkit';
import dataReducer from '../reducer/data/reducer';

export const store = configureStore({
  reducer: {
    dataPicker: dataReducer
  },
});
