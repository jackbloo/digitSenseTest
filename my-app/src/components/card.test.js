import React from 'react';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux';
import { store } from '../app/store';
import Card from './card'


describe('Unit Test for Card', () => {
    const data = {
        image: 'test.jpeg',
        title: 'test',
        content: 'test',
        id:2
    }
    test('Should be in the document', () => {
     render(
          <Provider store={store} >
            <Card data={data} changeNavigation={jest.fn()}/>
          </Provider>
        );
      const card = screen.getByTestId('card-container')
      expect(card).toBeInTheDocument();
      });

      test('Should match the snapshot', () => {
        render(
             <Provider store={store} >
               <Card data={data} changeNavigation={jest.fn()}/>
             </Provider>
           );
         const card = screen.getByTestId('card-container')
         expect(card).toMatchSnapshot()
         });
})

