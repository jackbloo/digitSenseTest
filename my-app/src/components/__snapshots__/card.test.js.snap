// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test for Card Should match the snapshot 1`] = `
<div
  class="sc-bdvvtL sOPvX"
  data-testid="card-container"
>
  <div
    class="sc-gsDKAQ"
  >
    <div
      class="sc-dkPtRN djYpFh"
    >
      <img
        alt="test"
        src="test.jpeg"
      />
    </div>
  </div>
  <div
    class="sc-hKwDye hrafIl"
  >
    <div>
      <div
        class="sc-eCImPb ifQwid"
      >
        test
      </div>
      <div
        class="sc-jRQBWg NnOMb"
      >
        test
      </div>
    </div>
    <div
      class="sc-gKclnd gGmwbb"
    >
      <div
        class="sc-iCfMLu coQalB"
      >
        Click for More
      </div>
      <div>
         Author: 
      </div>
    </div>
  </div>
</div>
`;
